src/BenchmarksServer/ProcessUtil.cs

```diff
@@ -12,7 +12,7 @@ namespace BenchmarkServer
     public static class ProcessUtil
     {
         public static ProcessResult Run(string filename, string arguments, string workingDirectory = null,
-            bool throwOnError = true, IDictionary<string, string> environmentVariables = null)
+            bool throwOnError = true, IDictionary<string, string> environmentVariables = null, int timeout = int.MaxValue)
```

Comment:
I would use `TimeSpan? timeout = null` instead of `int timeout = int.MaxValue`.  If `timeout` is null, call `WaitForExit()`, else call `WaitForExit(timeout.TotalMilliseconds)`.